Sources/SourceKitLSP/Swift/CodeActions/ApplyDeMorganLaw.swift

```diff
@@ -0,0 +1,551 @@
+      )
+    else {
+      return nil
+    }
+
+    binaryOptrExpr.tokenKindText = reversedBinaryOptrExprText
+    infixOptrExpr.operator = ExprSyntax(binaryOptrExpr)
+    infixOptrExpr.leftOperand = truthyLeftOperand
+    infixOptrExpr.rightOperand = truthyRightOperand
+
+    return ExprSyntax(infixOptrExpr)
+  }
+
+  func negatedToFalsy(
+    negationPrefix: String,
+    binaryOptrExprText: String,
+    reversedBinaryOptrExprText: String
+  )
+    -> ExprSyntax?
+  {
+    guard var infixOptrExpr = InfixOperatorExprSyntax(self),
+      var binaryOptrExpr = BinaryOperatorExprSyntax(infixOptrExpr.operator),
+      binaryOptrExpr.tokenKindText == binaryOptrExprText
+    else {
+      return untuplifyWithNegationHoisted { singleChild in
+        if var infixOptrExpr = InfixOperatorExprSyntax(singleChild),
+          var binaryOptrExpr = BinaryOperatorExprSyntax(infixOptrExpr.operator)
+        {
+          let reversedBinaryOptrExprText: String
+          switch binaryOptrExpr.tokenKindText {
+          case "==":
+            reversedBinaryOptrExprText = "!="
+          case "===":
+            reversedBinaryOptrExprText = "!=="
+          case _:
+            return (singleChild, negationPrefix)
+          }
+
+          binaryOptrExpr.tokenKindText = reversedBinaryOptrExprText
+          infixOptrExpr.operator = ExprSyntax(binaryOptrExpr)
+          return (ExprSyntax(infixOptrExpr), nil)
+        } else if let prefixOptrExpr = PrefixOperatorExprSyntax(singleChild),
+          prefixOptrExpr.tokenKindText == negationPrefix
+        {
+          return nil
+        } else {
+          return (singleChild, negationPrefix)
+        }
+      }
+    }
+
+    guard
+      let falsyLeftOperand = infixOptrExpr.leftOperand.negatedToFalsy(
+        negationPrefix: negationPrefix,
+        binaryOptrExprText: binaryOptrExprText,
+        reversedBinaryOptrExprText: reversedBinaryOptrExprText
+      ),
+      let falsyRightOperand = infixOptrExpr.rightOperand.negatedToFalsy(
+        negationPrefix: negationPrefix,
+        binaryOptrExprText: binaryOptrExprText,
+        reversedBinaryOptrExprText: reversedBinaryOptrExprText
+      )
+    else {
+      return nil
+    }
+
+    binaryOptrExpr.tokenKindText = reversedBinaryOptrExprText
+    infixOptrExpr.operator = ExprSyntax(binaryOptrExpr)
+    infixOptrExpr.leftOperand = falsyLeftOperand
+    infixOptrExpr.rightOperand = falsyRightOperand
+
+    return ExprSyntax(infixOptrExpr)
+  }
+}
+
+fileprivate extension PrefixOperatorExprSyntax {
+  var tokenKindText: String {
+    get {
+      if case let .prefixOperator(prefix) = `operator`.tokenKind {
+        prefix
+      } else {
+        ""
+      }
+    }
+    set(value) {
+      `operator`.tokenKind = .binaryOperator(value)
+    }
+  }
+
+  func appliedDeMorgan() -> ExprSyntax? {
+    let negationPrefix = tokenKindText
+    guard negationPrefix == "!" || negationPrefix == "~" else {
+      return nil
+    }
```

Comment:
Instead of passing arbitrary strings around for the `negationPrefix`, would it make sense to declare an enum for the two kinds of negation prefixes that are supported?